### PR TITLE
DEVPROD-5931: Support filtering by event type on host event logs

### PR DESCRIPTION
**Pull request by minnakt**
DEVPROD-5931

### Description
Adds filtering by event type for host events. Uses newer GraphQL query and new GraphQL enums. 

Although there are 31 different host events, most hosts in practice only have about 4~8 different types of events.

### Screenshots

Screenshot 2024-09-19 at 8 28 21 PM

### Testing
- Added Cypress tests
- Can test manually against production since the backend code has been deployed

### Evergreen PR
* evergreen-ci/evergreen/pull/8309

### CODE REVIEW

File: apps/spruce/src/pages/host/Metadata.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ const { gray } = palette;
 export const Metadata: React.FC<{
   loading: boolean;
   host: HostQuery["host"];
-  error: ApolloError;
+  error?: ApolloError;
```
Comment (SupaJoon) [inline apps/spruce/src/pages/host/Metadata.tsx:21]: nit: this is more aligned with actual usage
```suggestion
  error: ApolloError | undefined;
```

File: apps/spruce/src/pages/host/index.tsx [hunk 2]
```diff
@@ -23,17 +23,21 @@ import {
...
 import { Metadata } from "pages/host/Metadata";
 import { HostStatus } from "types/host";
+import { HostQueryParams } from "./constants";
 
 const Host: React.FC = () => {
   const dispatchToast = useToastContext();
   const { [slugs.hostId]: hostId } = useParams();
-  // Query host data
+
+  // Query host data.
```
Comment (SupaJoon) [inline apps/spruce/src/pages/host/index.tsx:40]: nit:
```suggestion
```

File: apps/spruce/src/pages/host/index.tsx [hunk 3]
```diff
@@ -55,27 +59,36 @@ const Host: React.FC = () => {
...
+    (bootstrapMethod === "ssh" || bootstrapMethod === "user-data");
 
   const { limit, page } = usePagination();
-  // Query hostEvent data
+  const [eventTypes] = useQueryParam<HostEventType[]>(
+    HostQueryParams.EventType,
+    [],
+  );
+
+  // UPDATE STATUS MODAL VISIBILITY STATE
```
Comment (SupaJoon) [inline apps/spruce/src/pages/host/index.tsx:73]: nit:
```suggestion
```

File: apps/spruce/src/pages/host/index.tsx [hunk 4]
```diff
@@ -55,27 +59,36 @@ const Host: React.FC = () => {
...
+  const [isUpdateStatusModalVisible, setIsUpdateStatusModalVisible] =
+    useState<boolean>(false);
+
+  // Query host event data.
```
Comment (SupaJoon) [inline apps/spruce/src/pages/host/index.tsx:77]: nit:
```suggestion
```

File: apps/spruce/src/pages/host/HostTable.tsx [hunk 5]
```diff
@@ -42,6 +75,18 @@ export const HostTable: React.FC<{
     });
   };
 
+  const eventTypeFilterOptions = useMemo(
+    () =>
+      eventTypes.map((e) => ({
+        title: e.toString().replaceAll("_", " "),
```
[the remark this answers is not in the document]
Comment (minnakt) [reply apps/spruce/src/pages/host/HostTable.tsx]: updated the filter text

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**